pandas/tests/arrays/string_/test_string.py

```diff
@@ -7,10 +7,54 @@
 
 import pandas as pd
 import pandas._testing as tm
+from pandas.core.arrays.string_arrow import ArrowStringArray, ArrowStringDtype
 
+skip_if_no_pyarrow = td.skip_if_no("pyarrow", min_version="1.0.0")
 
-def test_repr():
-    df = pd.DataFrame({"A": pd.array(["a", pd.NA, "b"], dtype="string")})
+
+@pytest.fixture(
+    params=[
+        # pandas\tests\arrays\string_\test_string.py:16: error: List item 1 has
+        # incompatible type "ParameterSet"; expected
+        # "Sequence[Collection[object]]"  [list-item]
+        "string",
+        pytest.param(
+            "arrow_string", marks=skip_if_no_pyarrow
+        ),  # type:ignore[list-item]
```

Comment:
Why is mypy failing here? (it seems we have such combination of a value with a pytest.param in many other places as well)

Reply:
I think we decided that'll we'll ignore these inference issues in tests rather than jump through hoops to appease mypy

does occur elsewhere..
https://github.com/pandas-dev/pandas/blob/792cc46588f880c457018bde1b60864e7b4020ad/pandas/tests/window/conftest.py#L77-L83

Reply:
Yeah, my question was more why the heck mypy *needs* the be pleased here .. ;)

But anyway, fully agree it is not worth going through hoops here if it is needed, but then let's just add the `# type:ignore`, and not all the lines with the error message from mypy, as that is rather distracting IMO

Reply:
I'd rather keep the message. If you don't want the message can you open an issue for that conversation. see #37715 and #37556

Reply:
But is this specific case fixable?

Reply:
that's why #37715 has been opened.. to get more eyeballs on these. if it is a mypy issue then we should have a mypy issue ref. if not the message could help someone with fixing.

I have added the messages in lots of places and don't want to change one case here. If you strongly disagree with this approach please open an issue. I'll then have more to say to justify this.